ExadelTimeTrackingSystem/ExadelTimeTrackingSystem.Data/Repositories/TaskRepository.cs

```diff
@@ -48,5 +59,40 @@ public Task BulkCreateAsync(List<Models.Task> tasks, CancellationToken cancellat
             cancellationToken.ThrowIfCancellationRequested();
             return GetCollection<Models.Task>().InsertManyAsync(tasks);
         }
+
+        public Task<List<string>> GetHours(DateTime date, Guid employeeId, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            List<DateTime> dateList = new List<DateTime>();
+            List<string> testList = new List<string>();
+            int total = 0;
+
+            for (int i = 0; i < 7; i++)
+            {
+                dateList.Add(date);
+                date = date.AddDays(1);
+            }
+
+            var filterBuilder = Builders<Models.Task>.Filter;
+            var employeeFilter = filterBuilder.Eq(t => t.EmployeeId, employeeId);
+            var datefilter = filterBuilder.In(t => t.Date, dateList);
+
+            var hours = GetCollection<Models.Task>().Aggregate().Match(datefilter & employeeFilter).Group(
+                t => t.Date,
+                group => new
+                {
+                    date = group.Key,
+                    Total = group.Sum(t => t.HoursSpent),
+                }).ToList();
+
+            foreach (var hour in hours)
+            {
+                total += hour.Total;
+                testList.Add(hour.date.DayOfWeek + ": " + hour.Total);
```

Comment:
combining results like day of week and ours and so on it's business logic so should be performed in service